pkg/controllers/raycluster_controller.go

```diff
@@ -619,3 +625,45 @@ func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return controller.Complete(r)
 }
+
+func deleteDeprecatedObjects(ctx context.Context, r *RayClusterReconciler, cluster *rayv1.RayCluster) error {
+	// Delete deprecated objects if they exist. These have all been replace by objects with names generated by
```

Comment:
Maybe add a TODO to have that logic removed after a reasonable number of releases?